back/engines/free/user_confirmation/spec/jobs/user_confirmation/expire_confirmation_code_job_spec.rb

```diff
@@ -9,8 +9,15 @@
     SettingsService.new.activate_feature! 'user_confirmation'
   end
 
-  it 'changes the confirmation code' do
+  it 'changes the confirmation code of an existing user record' do
+    old_code = user.email_confirmation_code
+    user.save!
+    described_class.perform_now(user.id, user.email_confirmation_code)
+    expect(user.reload.email_confirmation_code).not_to eq(old_code)
+  end
+
+  it 'does nothing when user record does not exist' do
     expect { described_class.perform_now(user, user.email_confirmation_code) }
-      .to(change(user, :email_confirmation_code))
+      .not_to(change(user, :email_confirmation_code))
```

Comment:
This is not a great test, as it doesn't test for the absence of the error we are avoiding. However, previously (before this fix) it would have failed as the Job would happily change the `email_confirmation_code` of the `User` _object_ it was passed, which is actually what is tested here. It would then subsequently have failed to save the updated object as a record, in the case the user record had been deleted, which would then raise the error we now avoid.

On balance, I feel this test, in combination with the new test, at least communicates something useful, and should catch the case where someone reverts / bypasses the changes in this PR.

Reply:
I agree that it communicates something useful.